### ISSUE TITLE
EndOfStreamException during journal recovery if journal file is zero bytes

### ISSUE THREAD

**Issue by Threetwosevensixseven** [mbdavid/LiteDB#162]
I discovered this by (ab)using Visual Studio Stop Debugging (Shift+F5) while doing inserts and updates in a loop.

In a fairly high percentage of cases, randomly killing the process during writes leaves a journal file zero bytes long. Next time you call `LiteDatabase` constructor it attempts journal recovery as usual, but fails with `EndOfStreamException`.

```
System.IO.EndOfStreamException was unhandled
  HResult=-2147024858
  Message=Unable to read beyond the end of the stream.
  Source=LiteDB
  StackTrace:
       at LiteDB.IOExceptionExtensions.WaitIfLocked(IOException ex, Int32 timer)
       at LiteDB.RecoveryService.OpenExclusiveFile(String filename, Action`1 success)
       at LiteDB.LiteDatabase..ctor(String connectionString)
       at TestApp.Program.Main(String[] args) in c:\Users\Robin\Documents\Visual Studio 2010\Projects\TestApp\TestApp\Program.cs:line 22
```

For now I'm using this helper to avoid the exception:

        public static void CleanupJournal()
        {
            var file = LiteDbName.Replace(".db", "-journal.db");
            if (!File.Exists(file)) return;
            var fi = new FileInfo(file);
            if (fi.Length == 0) fi.Delete();
        }

**Comment by mbdavid** [comment]
Hi `short.MaxValue`, this is strange. When LiteDB creates journal file the will write just after created. Yes, your process can brake after create journal file and before write first page on disk. If this occurs, is safe to delete journal file because datafile was not changed yet.
I will add this test when LiteDB try recover this empty datafile.

**Comment by Threetwosevensixseven** [comment]
Thanks `mbdavid`. 

`short.MaxValue`, I like it :D

**Comment by tjmoore** [comment]
Hi, is this considered fixed in 2.0.4?

**Comment by tjmoore** [comment]
Hi, sorry to bump, but I need to verify if this is fixed in 2,0.4 as have customers running into this issue in previous version of our software that is using 1.0.4 LiteDb and we're about to release a version that uses 2.0.4.

**Comment by mbdavid** [comment]
Hi @tjmoore, sorry for long time to anwser. This problem was fixed in 2.0.4 version. You can run some tests to check (this journal close mechanism was changed in v2 and v3)

**Comment by tjmoore** [comment]
Thanks. Actually just as posting I thought to simply check by creating a dummy zero length journal, and sure enough the file is removed and no crash.
